pkg/adapterManager/parallelManager_test.go

```diff
@@ -55,12 +56,12 @@ func TestExecute(t *testing.T) {
 			{&configpb.Adapter{Name: c.name}, &configpb.Aspect{Kind: c.name}},
 		}
 
-		o, err := m.Execute(context.Background(), cfg, nil, nil)
-		if c.inErr != nil && err == nil {
-			t.Errorf("m.Execute(...) = %v; want err: %v", err, c.inErr)
+		o := m.Execute(context.Background(), cfg, nil, nil)
```

Comment:
The problem then becomes "how do I map an err to a status?". I think the better way to think of this is that we're returning an error in the API domain rather than the golang domain. That being said, I think it's still reasonable for our aspect interface's return type to be `(Output, err)` so that adapters can differentiate between "I processed the user's request and rejected it" (API err) and "things are broken, I can't do any work" (golang err).